public/pages/DefineDetector/components/Datasource/DataSource.tsx

```diff
@@ -99,16 +195,45 @@ export function DataSource(props: DataSourceProps) {
     }
   };
 
-  const isDifferentIndex = () => {
-    return props.isEdit && indexName !== props.origIndex;
+  const isSelectedOptionIndexRemoved = (
+    newSelectedOptions: { label: string }[] = indexNames,
+    oldSelectedOptions: { label: string }[] = props.formikProps.values.index
+  ) => {
+    const newSelectedOptionsSet = new Set(newSelectedOptions);
+    const indexRemoved: boolean =
+      !(newSelectedOptions && oldSelectedOptions) ||
```

Comment:
If both newSelectedOptions and oldSelectedOptions are null or undefined, would it make more sense for isSelectedOptionIndexRemoved to return false instead of true, because if both are null or undefined, there is no change in the selection state — technically, no option was "removed." ?

Reply:
Originally if both were null it was okay for me to set it to true because then it would also make sense to reset the model configuration if we can't find any selected indices. 
I think actually though we pretty much never hit the case where both are undefined so it wasn't making too much of an effect and it logically makes more sense to say indexRemoved is false in that case and if we got to this point we would have had an indexRemoved previously.
I tested it out and removing `!(newSelectedOptins && oldSelectedOptions)` and adding:
```
    if (_.isEmpty(oldSelectedOptions) && _.isEmpty(newSelectedOptions)) {
      return false;
    }
```
also works

Reply:
make sense